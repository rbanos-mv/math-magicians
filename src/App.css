body {
  font-size: 30px;
}

.grid-container {
  display: grid;
  grid-template-columns: repeat(4, 100px);
  grid-template-rows: repeat(6, 75px);
  height: 450px;
  width: 400px;
}

#root {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.digit,
.operation {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.digit {
  background-color: lightgray;
  border: 1px solid gray;
}

.operation {
  background-color: orange;
  border: 1px solid gray;
}

/* Grey buttons */
#n0 {
  grid-column: 1 / 3;
  grid-row: 6;
}

#n1 {
  grid-column: 1;
  grid-row: 5;
}

#n2 {
  grid-column: 2;
  grid-row: 5;
}

#n3 {
  grid-column: 3;
  grid-row: 5;
}

#n4 {
  grid-column: 1;
  grid-row: 4;
}

#n5 {
  grid-column: 2;
  grid-row: 4;
}

#n6 {
  grid-column: 3;
  grid-row: 4;
}

#n7 {
  grid-column: 1;
  grid-row: 3;
}

#n8 {
  grid-column: 2;
  grid-row: 3;
}

#n9 {
  grid-column: 3;
  grid-row: 3;
}

#all-clear {
  grid-column: 1;
  grid-row: 2;
}

#plus-minus {
  grid-column: 2;
  grid-row: 2;
}

#percent {
  grid-column: 3;
  grid-row: 2;
}

#point {
  grid-column: 3;
  grid-row: 6;
}

/* Orange buttons */

#addition {
  grid-column: 4;
  grid-row: 5;
}

#division {
  grid-column: 4;
  grid-row: 2;
}

#equal {
  grid-column: 4;
  grid-row: 6;
}

#multiplication {
  grid-column: 4;
  grid-row: 3;
}

#subtraction {
  grid-column: 4;
  grid-row: 4;
}

#result {
  grid-column: 1 / 5;
  grid-row: 1;
  background-color: rgb(109, 108, 108);
  color: white;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding-right: 16px;
}
