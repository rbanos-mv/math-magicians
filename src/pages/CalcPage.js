import React from 'react';
import Calculator from '../components/Calculator';

const CalcPage = () => (
  <div className="calcpage">
    <h2>Let&apos;s do some math!</h2>
    <Calculator previous="" operand="" current="" solved={false} />
  </div>
);

export default CalcPage;
