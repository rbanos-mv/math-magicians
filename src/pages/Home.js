import React from 'react';

const Home = () => (
  <div className="home">
    <h2>Welcome to our page!</h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
      labore et dolore magna aliqua. Ultricies mi quis hendrerit dolor magna eget. Adipiscing elit
      pellentesque habitant morbi tristique senectus. Velit euismod in pellentesque massa placerat
      duis ultricies lacus. Amet facilisis magna etiam tempor orci eu lobortis elementum nibh.
      Tellus elementum sagittis vitae et leo duis ut diam quam. Porttitor massa id neque aliquam
      vestibulum. Enim diam vulputate ut pharetra sit. Enim eu turpis egestas pretium aenean
      pharetra. Cras tincidunt lobortis feugiat vivamus at augue eget. Duis convallis convallis
      tellus id interdum velit laoreet id donec.
    </p>
    <p>
      In hac habitasse platea dictumst vestibulum. Aliquet porttitor lacus luctus accumsan.
      Scelerisque purus semper eget duis at tellus at. In fermentum et sollicitudin ac orci
      phasellus egestas tellus. Cursus turpis massa tincidunt dui ut ornare. Id leo in vitae turpis
      massa sed elementum tempus egestas. Vestibulum mattis ullamcorper velit sed ullamcorper morbi
      tincidunt ornare massa. Ultrices eros in cursus turpis massa tincidunt dui. Iaculis at erat
      pellentesque adipiscing commodo elit. Vitae ultricies leo integer malesuada nunc vel. Massa
      massa ultricies mi quis hendrerit dolor magna. Nibh tortor id aliquet lectus proin nibh nisl.
      Est sit amet facilisis magna etiam tempor orci eu. Ut sem viverra aliquet eget sit amet
      tellus. Pharetra massa massa ultricies mi quis hendrerit dolor magna.
    </p>
  </div>
);

export default Home;
