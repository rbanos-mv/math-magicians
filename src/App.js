import React from 'react';
import {
  HashRouter as Router, Routes, Route, Link,
} from 'react-router-dom';
import './App.css';
import Home from './pages/Home';
import CalcPage from './pages/CalcPage';
import Quote from './pages/Quote';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Router>
        <header>
          <h1>Math Magicians</h1>
          <div className="nav">
            <nav>
              <Link className="link" to="/">
                Home
              </Link>
              <Link className="link" to="/calculator">
                Calculator
              </Link>
              <Link className="link" to="/quote">
                Quote
              </Link>
            </nav>
          </div>
        </header>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/calculator" element={<CalcPage />} />
          <Route path="/quote" element={<Quote />} />
          <Route path="*" element={<div>Page not found</div>} />
        </Routes>
      </Router>
    );
  }
}

export default App;
