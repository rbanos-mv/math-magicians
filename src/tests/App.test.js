import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import App from '../App';
import Home from '../pages/Home';
import CalcPage from '../pages/CalcPage';

describe('Test the App components', () => {
  it('renders the App correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Home page correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Calculator page correctly', () => {
    const tree = renderer.create(<CalcPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows the home page when clicking the navbar link', () => {
    const { container } = render(<App url="/" />);
    fireEvent.click(screen.getByText('Calculator'));
    fireEvent.click(screen.getByText('Home'));
    expect(container.getElementsByClassName('home')[0]).toHaveTextContent('Welcome to our page!');
  });

  it('shows the calculator page when clicking the navbar link', () => {
    const { container } = render(<App url="/" />);
    fireEvent.click(screen.getByText('Calculator'));
    expect(container.getElementsByClassName('calcpage')[0]).toHaveTextContent(
      "Let's do some math!",
    );
  });

  it('shows the quote page when clicking the navbar link', () => {
    const { container } = render(<App url="/" />);
    fireEvent.click(screen.getByText('Quote'));
    expect(container.getElementsByClassName('quote')[0]).toBeTruthy();
  });

  it('performs math operation on calculator', () => {
    const { container } = render(<CalcPage />);
    expect(container.getElementsByClassName('calcpage')[0]).toHaveTextContent(
      "Let's do some math!",
    );

    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('Ã—'));
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('='));
    expect(container.getElementsByClassName('previous')[0]).toHaveTextContent('63');
  });
});
