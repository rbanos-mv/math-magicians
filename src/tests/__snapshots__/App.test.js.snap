// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App components renders the App correctly 1`] = `
Array [
  <header>
    <h1>
      Math Magicians
    </h1>
    <div
      className="nav"
    >
      <nav>
        <a
          className="link"
          href="#/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="link"
          href="#/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <a
          className="link"
          href="#/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </nav>
    </div>
  </header>,
  <div
    className="home"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ultricies mi quis hendrerit dolor magna eget. Adipiscing elit pellentesque habitant morbi tristique senectus. Velit euismod in pellentesque massa placerat duis ultricies lacus. Amet facilisis magna etiam tempor orci eu lobortis elementum nibh. Tellus elementum sagittis vitae et leo duis ut diam quam. Porttitor massa id neque aliquam vestibulum. Enim diam vulputate ut pharetra sit. Enim eu turpis egestas pretium aenean pharetra. Cras tincidunt lobortis feugiat vivamus at augue eget. Duis convallis convallis tellus id interdum velit laoreet id donec.
    </p>
    <p>
      In hac habitasse platea dictumst vestibulum. Aliquet porttitor lacus luctus accumsan. Scelerisque purus semper eget duis at tellus at. In fermentum et sollicitudin ac orci phasellus egestas tellus. Cursus turpis massa tincidunt dui ut ornare. Id leo in vitae turpis massa sed elementum tempus egestas. Vestibulum mattis ullamcorper velit sed ullamcorper morbi tincidunt ornare massa. Ultrices eros in cursus turpis massa tincidunt dui. Iaculis at erat pellentesque adipiscing commodo elit. Vitae ultricies leo integer malesuada nunc vel. Massa massa ultricies mi quis hendrerit dolor magna. Nibh tortor id aliquet lectus proin nibh nisl. Est sit amet facilisis magna etiam tempor orci eu. Ut sem viverra aliquet eget sit amet tellus. Pharetra massa massa ultricies mi quis hendrerit dolor magna.
    </p>
  </div>,
]
`;

exports[`Test the App components renders the Calculator page correctly 1`] = `
<div
  className="calcpage"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="grid-container"
  >
    <div
      id="result"
    >
      <span
        className="previous"
      />
       
      <span
        className="operand"
      />
       
      <span
        className="current"
      />
    </div>
    <button
      className="digit"
      id="all-clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="digit"
      id="plus-minus"
      onClick={[Function]}
      type="button"
    >
      ±
    </button>
    <button
      className="digit"
      id="mod"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operation"
      id="division"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="digit"
      id="n7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="digit"
      id="n8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="digit"
      id="n9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operation"
      id="multiplication"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
    <button
      className="digit"
      id="n4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="digit"
      id="n5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="digit"
      id="n6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operation"
      id="subtraction"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="digit"
      id="n1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="digit"
      id="n2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="digit"
      id="n3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operation"
      id="addition"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="digit"
      id="n0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="digit"
      id="point"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operation"
      id="equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Test the App components renders the Home page correctly 1`] = `
<div
  className="home"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ultricies mi quis hendrerit dolor magna eget. Adipiscing elit pellentesque habitant morbi tristique senectus. Velit euismod in pellentesque massa placerat duis ultricies lacus. Amet facilisis magna etiam tempor orci eu lobortis elementum nibh. Tellus elementum sagittis vitae et leo duis ut diam quam. Porttitor massa id neque aliquam vestibulum. Enim diam vulputate ut pharetra sit. Enim eu turpis egestas pretium aenean pharetra. Cras tincidunt lobortis feugiat vivamus at augue eget. Duis convallis convallis tellus id interdum velit laoreet id donec.
  </p>
  <p>
    In hac habitasse platea dictumst vestibulum. Aliquet porttitor lacus luctus accumsan. Scelerisque purus semper eget duis at tellus at. In fermentum et sollicitudin ac orci phasellus egestas tellus. Cursus turpis massa tincidunt dui ut ornare. Id leo in vitae turpis massa sed elementum tempus egestas. Vestibulum mattis ullamcorper velit sed ullamcorper morbi tincidunt ornare massa. Ultrices eros in cursus turpis massa tincidunt dui. Iaculis at erat pellentesque adipiscing commodo elit. Vitae ultricies leo integer malesuada nunc vel. Massa massa ultricies mi quis hendrerit dolor magna. Nibh tortor id aliquet lectus proin nibh nisl. Est sit amet facilisis magna etiam tempor orci eu. Ut sem viverra aliquet eget sit amet tellus. Pharetra massa massa ultricies mi quis hendrerit dolor magna.
  </p>
</div>
`;
